iceoryx_posh/source/mepoo/memory_manager.cpp

```diff
@@ -178,14 +178,15 @@ SharedChunk MemoryManager::getChunk(const MaxSize_t f_size)
         printMemPoolVector();
         std::cerr << "\nCould not find a fitting mempool for a chunk of size " << adjustedSize << std::endl;
 
-        errorHandler(Error::kMEPOO__MEMPOOL_GETCHUNK_CHUNK_IS_TOO_LARGE);
+        errorHandler(Error::kMEPOO__MEMPOOL_GETCHUNK_CHUNK_IS_TOO_LARGE, nullptr, ErrorLevel::SEVERE);
         return SharedChunk(nullptr);
     }
     else if (chunk == nullptr)
     {
         std::cerr << "MemoryManager: unable to acquire a chunk with a payload size of " << f_size << std::endl;
         std::cerr << "The following mempools are available:" << std::endl;
         printMemPoolVector();
+        errorHandler(Error::kMEPOO__MEMPOOL_GETCHUNK_POOL_IS_RUNNING_OUT_OF_CHUNKS, nullptr, ErrorLevel::MODERATE);
```

Comment:
this reminds me that we need to port some parts of out code base to `cxx::expected`. When the publisher uses `cxx::expected` for the `load` then the lower layers should probably also use it.

Reply:
@elBoberido Is there an issue or should I create one?

Reply:
There is no issue as far as I know. This would fit into a generic cleanup issue. There is also other stuff, like `noexcept` is not used everywhere.

Reply:
Created #419 and #420.